lib/riemann.js

```diff
@@ -9,8 +9,36 @@ var riemann = require("riemann");
  * Optional:
  * transport, flush, packet
  */
+
+
 function Riemann(startupTime, config, emitter) {
   this.config = config.riemann;
+  this.emitter = emitter;
+
+  if (this.config.transport && this.config.transport === "tcp") {
+    setInterval(this.reconnectIfNeeded.bind(this), this.config.reconnectInterval);
+  } else {
+    this.setupRiemannClient();
+  }
+}
+
+Riemann.prototype.reconnectIfNeeded = function() {
+    if (!this.client) {
+        if (this.config.debug) {
+            console.log('Connecting to Reimann');
+        }
+
+        this.setupRiemannClient();
+    } else if (this.client.tcp.socket.readyState == 'closed') {
```

Comment:
It would probably be a good idea to protect this lookup incase any of the components are undefined.